addons/account/models/sequence_mixin.py

```diff
@@ -60,10 +82,25 @@ def _deduce_sequence_number_reset(self, name):
         """
         def _check_grouping(grouping, optional=None, required=None):
             sequence_dict = sequence.groupdict()
-            check = all(key in sequence_dict for key in (optional or [])) and all(sequence_dict.get(key) for key in (required or []))
-            check &= 'year' not in (required or []) or 2000 <= int(sequence_dict.get('year', -1)) <= 2100 or len(sequence_dict.get('year', '')) == 2
-            check &= 'month' not in (required or []) or 1 <= int(sequence_dict.get('month', -1)) <= 12
-            return check
+            optional = optional or []
+            required = required or []
+            if not (
+                all(key in sequence_dict for key in optional)
+                and all(sequence_dict.get(key) for key in required)
+            ):
+                return False
+            if not (
+                'year' not in required
+                or 2000 <= int(sequence_dict.get('year') or -1) <= 2100
+                or len(sequence_dict.get('year') or '') == 2
+            ):
+                return False
+            if not (
+                'month' not in required
+                or 1 <= int(sequence_dict.get('month') or -1) <= 12
+            ):
```

Comment:
`if 'month' in required and not (1 <= int(sequence_dict.get('month') or -1) <= 12)` ?